scripts/build.ts

```diff
@@ -20,102 +25,172 @@ function genChainMetadataMapExport(data, exportName) {
+function createTmpDir() {
+  console.log('Preparing tmp directory');
+  if (fs.existsSync('./tmp')) fs.rmSync(`./tmp`, { recursive: true });
+  // Start with the contents of src, which we will add to in this script
+  fs.cpSync(`./src`, `./tmp`, { recursive: true });
+}
+
+function createChainFiles() {
+  console.log('Parsing and copying chain data');
+  for (const file of fs.readdirSync('./chains')) {
+    const inDirPath = `./chains/${file}`;
+    const assetOutPath = `./dist/chains/${file}`;
+    // ts files go to the tmp dir so they can be compiled along with other generated code
+    const tsOutPath = `./tmp/chains/${file}`;
+    const stat = fs.statSync(`${inDirPath}`);
+    if (!stat.isDirectory()) continue;
+
+    // Convert and copy metadata
+    const metadata = parse(fs.readFileSync(`${inDirPath}/metadata.yaml`, 'utf8'));
+    chainMetadata[metadata.name] = metadata;
+    fs.mkdirSync(`${assetOutPath}`, { recursive: true });
+    fs.mkdirSync(`${tsOutPath}`, { recursive: true });
+    fs.copyFileSync(`${inDirPath}/metadata.yaml`, `${assetOutPath}/metadata.yaml`);
+    fs.writeFileSync(`${assetOutPath}/metadata.json`, JSON.stringify(metadata, null, 2));
+    fs.writeFileSync(`${tsOutPath}/metadata.ts`, genChainMetadataExport(metadata, 'metadata'));
+
+    // Convert and copy addresses if there are any
+    if (fs.existsSync(`${inDirPath}/addresses.yaml`)) {
+      const addresses = parse(fs.readFileSync(`${inDirPath}/addresses.yaml`, 'utf8'));
+      chainAddresses[metadata.name] = addresses;
+      fs.copyFileSync(`${inDirPath}/addresses.yaml`, `${assetOutPath}/addresses.yaml`);
+      fs.writeFileSync(`${assetOutPath}/addresses.json`, JSON.stringify(addresses, null, 2));
+      fs.writeFileSync(`${tsOutPath}/addresses.ts`, genJsExport(addresses, 'addresses'));
+    }
+
+    // Copy the logo file
+    fs.copyFileSync(`${inDirPath}/logo.svg`, `${assetOutPath}/logo.svg`);
   }
+}
+
+function createWarpConfigFiles() {
+  console.log('Parsing and copying warp config data');
+  const warpPathBase = 'deployments/warp_routes';
+  // Outer loop for token symbol directories
+  for (const warpDir of fs.readdirSync(`./${warpPathBase}`)) {
+    const inDirPath = `./${warpPathBase}/${warpDir}`;
+    const assetOutPath = `./dist/${warpPathBase}/${warpDir}`;
+    // ts files go to the tmp dir so they can be compiled along with other generated code
+    const tsOutPath = `./tmp/${warpPathBase}/${warpDir}`;
+    const stat = fs.statSync(`${inDirPath}`);
+    if (!stat.isDirectory()) continue;
 
-  // Copy the logo file
-  fs.copyFileSync(`${inDirPath}/logo.svg`, `${assetOutPath}/logo.svg`);
+    // Inner loop for individual warp route configs
+    for (const warpFile of fs.readdirSync(inDirPath)) {
+      if (!warpFile.endsWith('config.yaml')) continue;
+      const [warpFileName] = warpFile.split('.');
+      const config = parse(fs.readFileSync(`${inDirPath}/${warpFile}`, 'utf8'));
+      const id = warpRouteConfigToId(config);
```

Comment:
isnt this pretty similar to the filepath? whats the benefit here

Reply:
It is similar yes. The id could have been a file path but:
1. The file paths have some extra stuff in it (`-config.yaml`)
2. After having to work around slashes in cosmos IBC token denoms, I've become weary of using `/` as a separator symbol in our IDs

Reply:
Resolved offline, we'll keep these ID utilities but change the format a bit from `USDT|chain1-chain2` to `USDT/chain1-chain2`